aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -12,16 +12,21 @@ RegisterOperators& RegisterOperators::operator=(RegisterOperators&&) noexcept =
 class RegisterOperators::OperatorRegistrar final {
 public:
   explicit OperatorRegistrar(FunctionSchema&& schema, c10::optional<TensorTypeId> dispatch_key, KernelFunction* kernel, KernelCacheCreatorFunction&& cache_creator)
-  : op_(Dispatcher::singleton().registerSchema(std::move(schema))), dispatch_key_(std::move(dispatch_key)), owns_registration_(true) {
-    if (dispatch_key_.has_value()) {
-      Dispatcher::singleton().registerKernel(op_, *dispatch_key_, kernel, std::move(cache_creator));
-    } else {
-      Dispatcher::singleton().registerFallbackKernel(op_, kernel, std::move(cache_creator));
+  : op_(Dispatcher::singleton().registerSchema(std::move(schema))), dispatch_key_(std::move(dispatch_key)), has_kernel_(kernel != nullptr), owns_registration_(true) {
```

Comment:
nit: also could be optional<> instead of has_kernel

Reply:
Optional where? there is no kernel member I could make optional.